Plugins/NativeXr/Source/NativeXr.cpp

```diff
@@ -1411,6 +1517,72 @@ namespace Babylon
                     }
                 }
             }
+
+            Napi::Value GetWorldInformation(const Napi::CallbackInfo& info)
+            {
+                // Create a JavaScript object that stores all world information.
+                Napi::Object worldInformationObj = Napi::Object::New(info.Env());
+
+                // Create a set to contain all of the currently tracked planes.
+                Napi::Object planeSet = info.Env().Global().Get("Set").As<Napi::Function>().New({});
+
+                // Loop over the list of tracked planes, and add them to the set.
+                for (const auto& [plane, planeNapiValue] : m_trackedPlanes)
+                {
+                    planeSet.Get("add").As<Napi::Function>().Call(planeSet, {planeNapiValue.Value()});
+                }
+
+                // Pass the world information object back to the caller.
+                worldInformationObj.Set("detectedPlanes", planeSet);
+                return std::move(worldInformationObj);
+            }
+
+            void UpdatePlanes(const Napi::Env& env, uint32_t timestamp)
+            {
+                std::map<xr::NativePlaneIdentifier, xr::Plane*> existingNativePlaneMap;
```

Comment:
Can this be `xr::Plane&`? It's expected to never be `nullptr`, correct?

Reply:
You can't use xr::Plane& in a std::unordered_map as it always tries to copy it.

Reply:
Well I'll be damned nevermind, this does work for maps, but does not work for vectors.  I will use that here.